doc/loader.rst

```diff
@@ -5,128 +5,539 @@ MIG LOADER
+to that system.
+
+::
+
+    mig> create loader
+    Entering loader creation mode.
+    Please provide the name of the new entry
+    name> corbomite.internal
+    Name: 'corbomite.internal'
+    Please provide loader key for entry.
+    key> vw1NQs9F3wuZx1kSUhZaQzZ0
+    Key: 'vw1NQs9F3wuZx1kSUhZaQzZ0'
+    {
+      "ID": 0,
+      "Name": "corbomite.internal",
+      "Key": "vw1NQs9F3wuZx1kSUhZaQzZ0",
+      "AgentName": "",
+      "LastSeen": "0001-01-01T00:00:00Z",
+      "Enabled": false
+    }
+    create loader entry? (y/n)> y
+    New entry successfully created but is disabled
+    mig>
+
+The name can be any value you want, but usually you will want something describing
+the system or in the case of a workstation something describing the user of the
+device. Here we just used the hostname. The key is the API key that will need to be
+configured in mig-loader on that system to allow it to authenticate as this loader
+instance.
+
+The new loader is created in a disabled state. Lets enable it so that it can be
+used.
+
+::
+
+    mig> search loader where loadername=%corb%
+    Searching loader after 2011-11-05T20:22:49Z and before 2020-11-17T20:22:49Z, limited to 100 results
+    - ID - + ----      Name      ---- + ----   Agent Name   ---- + -- Enabled - + -- Last Used ---
+        12   corbomite.internal         unset                      false          2016-05-12T20:16:30Z
+    mig> loader 12
+    Entering loader reader mode. Type exit or press ctrl+d to leave. help may help.
+    Loader: 'corbomite.internal'.
+    Status 'false'.
+    loader 12> enable
+    Loader has been enabled
+    reloaded
+    loader 12>
+
+Note the agent name is unset as it has not been used yet. Once mig-loader connects
+and authenticates as this loader instance, it will be populated with the hostname of
+the device.
+
+Adding the new instance to the relay
+------------------------------------
+The loader instance configured using mig-console allows authentication with the
+API. We also need to add the credentials to RabbitMQ, as they will also be used by
+the loader provisioned agent to connect. This was mentioned previously with respect
+to using AMQPCRED in the agent configuration file.
+
+Add a new user to the MIG RabbitMQ vhost with the same key used for the loader
+instance. Ensure the correct permissions are set on the new RabbitMQ user. You
+can use ``rabbitmqctl`` for this, or use the RabbitMQ admin interface.
+
+The following example shows how you would add a user and set the required
+permissions on your relay, using ``rabbitmqctl``.
+
+::
+
+    # rabbitmqctl add_user corbomite.internal vw1NQs9F3wuZx1kSUhZaQzZ0
+    Creating user "corbomite.internal" ...
+    # rabbitmqctl set_permissions -p mig corbomite.internal '^mig\.agt\..*$' '^(toschedulers|mig\.agt\..*)$' '^(toagents|mig\.agt\..*)$'
+    Setting permissions for user "corbomite.internal" in vhost "mig" ...
+    #
```

Comment:
We still need to fix this. MIG Administrators won't have access to `rabbitmqctl`.